File: domain/src/main/java/de/x4fyr/timeline/adapter/ScheduleAdapter.java
package de.x4fyr.timeline.adapter;

import de.x4fyr.timeline.domain.elements.ScheduledElement;

import java.time.LocalDateTime;
import java.util.Collection;

/**
 * @author Benedikt Volkmer
 *         Created on 11/14/16.
 */
public interface ScheduleAdapter {

    ScheduledElement saveToSchedule(ScheduledElement element);

    void deleteFromSchedule(ScheduledElement element);

    Collection<ScheduledElement> getByTimeSpan(LocalDateTime start, LocalDateTime end);
}

File: domain/src/main/kotlin/de/x4fyr/timeline/adapter/ScheduleAdapter.kt
package de.x4fyr.timeline.adapter

import de.x4fyr.timeline.domain.elements.ScheduledElement
import org.joda.time.LocalDateTime


/**
 * @author Benedikt Volkmer
 *         Created on 11/14/16.
 */
interface ScheduleAdapter {

    /** Save the given ScheduleElement to the schedule persistence */
    fun saveToSchedule(element: ScheduledElement): ScheduledElement

    /** Delete given ScheduleElement from schedule persistence */
    fun deleteFromSchedule(element: ScheduledElement)

    /** get ScheduleElements of given time span */
    fun getByTimeSpan(start: LocalDateTime, end: LocalDateTime): Set<ScheduledElement>

    fun getNext(element: ScheduledElement, count: Long): Set<ScheduledElement>

    fun getPrev(element: ScheduledElement, count: Long): Set<ScheduledElement>
}
